improver/utilities/temporal.py

```diff
@@ -281,3 +281,69 @@ def relabel_to_period(cube: Cube, period: Optional[int] = None):
             dtype=TIME_COORDS[coord].dtype,
         )
     return cube
+
+
+def integrate_time(cube: Cube, new_name: str = None) -> Cube:
+    """
+    Multiply a frequency or rate cube by the time period given by the
+    time bounds over which it is defined to return a count or accumulation.
+    The frequency or rate must be defined with time bounds, e.g. an average
+    frequency across the period. This function will handle a cube with a
+    non-scalar time coordinate, multiplying each time in the coordiante by the
+    related bounds.
+
+    The returned cube has units equivalent to the input cube multiplied by
+    seconds.
+
+    Any time related cell methods are removed from the output cube and a new
+    "sum" over time cell method is added.
+
+    Args:
+        Cube:
+            A cube of average frequency or rate within a defined period.
+        new_name:
+            A new name for the resulting diagnostic.
+
+    Returns:
+        The cube with the data multiplied by the period in seconds defined
+        by the bounds on the time coordinate.
+
+    Raises:
+        ValueError: If the input cube time coordinate does not have time
+                    bounds.
+    """
+    # Ensure cube has a time coordinate with bounds
+    if not cube.coord("time").has_bounds():
+        raise ValueError(
+            "time coordinate must have bounds to apply this time-bounds integration"
+        )
+
```

Comment:
At the moment, I don't think anything is stopping users from providing a rate/frequency which is (for example) 'per hour', which would lead to unexpected results. 

Could an additional check be added along the lines of:
```suggestion
if str(cube.units).find('s-') == -1:
    raise ValueError('message')
```
A corresponding unit test would also be required to demonstrate this functionality.

Reply:
The code will modify the units of the returned cube to ensure that any input unit can be handled. The data will be integrated in seconds, but the unit will include a factor that accounts for the different input units, that why the unit is modified using the `* Unit("s")` approach. I had failed to demonstrate this in the units tests, so I've added a relevant test.